tests/test_precedence.rs

```diff
@@ -108,32 +116,32 @@ fn test_rustc_precedence() {
             }
             Err(msg) => {
                 errorf!("syn failed to parse\n{:?}\n", msg);
-                failed += 1;
                 (0, 1)
             }
         };
 
-        passed += l_passed;
-        failed += l_failed;
-
         errorf!("=== {}: {} passed | {} failed\n", path.display(), l_passed, l_failed);
 
-        if failed >= abort_after {
-            errorf!("Aborting Immediately due to ABORT_AFTER_FAILURE\n");
-            break;
+        passed.fetch_add(l_passed, Ordering::SeqCst);
+        let prev_failed = failed.fetch_add(l_failed, Ordering::SeqCst);
+
+        if prev_failed + 1 >= abort_after {
```

Comment:
I think you should have `prev_failed + l_failed` here, not `+ 1`, no?

Reply:
Wow, yes nice catch.